File: lib/database/student_dao.dart
import 'package:android_native_flutter_db/database/student_entity.dart';
import 'package:floor/floor.dart';

@dao
abstract class StudentDao {
  @Query('SELECT * FROM students')
  Future<List<Student>> findAllStudents();

  @Query('SELECT * FROM students WHERE id = :id')
  Stream<Student?> findStudentById(int id);

  @insert
  Future<void> insertStudent(List<Student> student);

  @Query('DELETE FROM students WHERE id = :id')
  Future<void> delete(int id);
}

File: android/app/src/main/kotlin/com/example/android_native_flutter_db/database/StudentDao.kt
package com.example.android_native_flutter_db.database

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface StudentDao {

    @Query("SELECT * FROM students")
    fun findAllStudents(): List<Student>

    @Query("SELECT * FROM students WHERE id = :id")
    fun findStudentById(id : Int): Flow<List<Student>>

    @Insert
    suspend fun insertStudent(student: Student)

    @Delete
    suspend fun deleteStudent(subscriber: Student): Int

}